src/glissade/preprocessing.py

```
@@ -28,27 +29,53 @@ def read_data(db_file : str, denovo_file : str):
     db_df['scan'] = db_scans
+    # Use the filename column for file identity (supports multi-file analyses)
+    db_df['file_stem'] = db_df['filename'].apply(lambda x: os.path.splitext(os.path.basename(str(x)))[0])
   else:
     #FIXME handle other search results
     pass
 
   if '.mztab' in denovo_file:
+    # Build ms_run -> file stem mapping from MTD header
+    run_map = {}
     with open(denovo_file) as f_in:
       for skiprows, line in enumerate(f_in):
```

Ruff (0.15.20)

[warning] Loop control variable `skiprows` not used within loop body

(B007)